tools/sortimports/main.go

```diff
@@ -0,0 +1,257 @@
+/*
+ * Copyright 2025 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package main
+
+import (
+	"bytes"
+	"flag"
+	"fmt"
+	"go/ast"
+	"go/format"
+	"go/parser"
+	"go/printer"
+	"go/token"
+	"io/fs"
+	"log"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+const (
+	// the below values taken from src/go/format/format.go
+	tabWidth                = 8
+	printerNormalizeNumbers = 1 << 30
+	printerMode             = printer.UseSpaces | printer.TabIndent | printerNormalizeNumbers
+)
+
+var config = printer.Config{Mode: printerMode, Tabwidth: tabWidth}
+
+type Args struct {
+	Paths       []string
+	ExcludeDirs sets.Set[string]
+	Verbose     bool
+}
+
+// Sorts imports based on:
+// 1. Standard libraries
+// 2. "k8s.io" imports
+// 3. "sigs.k8s.io" imports
+// 4. Other third-party libraries
+// 5. "github.com/openshift" imports
+// Custom sorting function for import specs
+func sortImports(imports []*ast.ImportSpec) []*ast.ImportSpec {
+	var stdLibs, k8sLibs, thirdParty, openShiftLibs []*ast.ImportSpec
+
+	for _, imp := range imports {
+		path := strings.Trim(imp.Path.Value, `"`)
+		switch {
+		case isStdLib(path):
+			stdLibs = append(stdLibs, imp)
+		case strings.HasPrefix(path, "k8s.io") || strings.HasPrefix(path, "sigs.k8s.io"):
+			k8sLibs = append(k8sLibs, imp)
+		case strings.HasPrefix(path, "github.com/openshift"):
+			openShiftLibs = append(openShiftLibs, imp)
```

Comment:
IIUC openshift and opnshift-kni will be grouped (and sorted later) together. I think the local imports of numaresources-operator project would look nicer if split from other openshift packages